graft/evm/firewood/recovery.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
```

Comment:
Is it possible to test this code? units tests for things like                                                                                                                                                   

- multiple children with same state root
- other edge cases like this 
- concurrent operations

could be useful